ast/parser.go

```diff
@@ -831,6 +831,10 @@ func (p *Parser) parseHead(defaultRule bool) (*Head, bool) {
 
 	switch p.s.tok {
 	case tokens.Contains: // NOTE: no Value for `contains` heads, we return here
+		// Catch error case of using 'contains' with a function definition rule head.
```

Comment:
This is located before the scan to have the error location land _on_ the keyword. Using `p.save()` + `p.restore()` later would have a similar effect.